kafka/README.md

```diff
@@ -33,11 +51,14 @@ This object contains information about the operation representation in Kafka.
 
 ##### Fixed Fields
 
-Field Name | Type | Description
----|:---:|---
-<a name="operationBindingObjectGroupId"></a>`groupId` | [Schema Object][schemaObject] | Id of the consumer group.
-<a name="operationBindingObjectClientId"></a>`clientId` | [Schema Object][schemaObject] | Id of the consumer inside a consumer group.
-<a name="operationBindingObjectBindingVersion"></a>`bindingVersion` | string | The version of this binding. If omitted, "latest" MUST be assumed.
+Field Name | Type | Description | Applicability [default] | Constraints
+---|:---:|:---:|:---:|---
+<a name="operationBindingObjectGroupId"></a>`groupId` | [Schema Object][schemaObject] | Id of the consumer group. | OPTIONAL | -
+<a name="operationBindingObjectClientId"></a>`clientId` | [Schema Object][schemaObject] | Id of the consumer inside a consumer group. | OPTIONAL | -
+<a name="operationBindingObjectSchemaIdLocation"></a>`schemaIdLocation` | string | If a Schema Registry is used when performing this operation, tells where the id of schema is stored (e.g. `header` or `payload`). | OPTIONAL | MUST NOT be specified if `schemaRegistryUrl` is not specified at the Server level
```

Comment:
If someone chooses `header` we might want a place to identify what the header is.

Reply:
Hi @dalelane
Is this header value configurable for some Ser/Deser libraries? I thought it was just just dependant on the `schemaRegistryVendor` and couldn't be configured...
If not configurable, I would stick with the header being deduced from the registry vendrod.
What do you think?

Reply:
Most schema registries support the Confluent API implementation (in addition to their own APIs), so it's fairly common for apps to use Confluent client libraries pointing at a non-Confluent schema registry, essentially treating it as running in a Confluent-compatibility-mode. The Apicurio schema registry even supports the IBM API and the Confluent API, as well as it's own. ((ref)[https://www.apicur.io/registry/docs/apicurio-registry/2.2.x/getting-started/assembly-intro-to-the-registry.html#_apicurio_registry_capabilities]) 

So I was thinking about how we document this mismatch - where you can't assume you know what a client application is doing just because you know the schema registry vendor. 

I'm actively nit-picking and looking for edge cases to make sure we have something that will cope with all scenarios... but I think your probably right to keep it simple with fewer options. For those scenarios, we can just advise devs to describe their schema registry vendor as how it is being used (e.g. if you're using an Apicurio schema registry with Confluent clients, taking advantage of the Confluent-compatibility APIs, then just describe your schema registry as being Confluent for the purposes of an AsyncAPI document)

Reply:
Keep it simple for the moment and propose to resolve it. Please re-open if any new insights